w07-ruby-on-rails-cont/d3-homework/README.md

```diff
@@ -0,0 +1,88 @@
+
+## Fork
+
+Forking is used when you are not a collaborator on a project. Just read the following scenario.
+
+Say you want to add some better test coverage to a wonderful and popular starter project for Express. You don't own the repo, and you haven't been added as a collaborator.  This is the most common way for individuals and companies to interact with open source projects' code on GitHub.  
+
+1. First, <a href="https://help.github.com/articles/fork-a-repo/" target="_blank">fork</a> the repo. This will create a duplicate remote repo in GitHub under your GitHub account. Then, clone the new repo from your GitHub account to your local computer.
+2. Now make a new branch called `more-test-coverage`. Add your tests. How will you get your changes back to the original project from a fork?
+3. It's not always necessary, but it's good practice to update your project from the original in case changes occurred in the original repo while you were working. This is the equivalent of pulling from origin/master before pushing, except now there are two different remote repositories on GitHub to manage. Add the original project remote as `upstream` by copying the clone address of the original repo.
+  ```
+  $ git remote add upstream <PASTE ORIGINAL GITHUB REPO CLONE URL HERE>
+  $ git checkout master
+  $ git pull upstream master
+  ```
+4. Merge the changes from local/master into your local/more-test-coverage branch.
+  ```
+  $ git checkout more-test-coverage
+  $ git merge master
+  ```
+4. When you are satisfied, push the branch to your origin (since you forked, this will be your forked repo, not the original project).
+  ```
+  $ git push origin more-test-coverage
+  ```
+5. Navigate your browser to the Github page of your remote. Github will recognize that this project is forked from another and ask if you would like to submit the `more-test-coverage` branch to the original project as a pull request. Say "oh yeah!"
+6. Whoever owns the repo for the starter express project will now receive an email saying that you submitted a pull request. They will be able to pull down your pull request branch to examine it and merge it into the project if appropriate. Watch for messages from the repo owner asking questions or suggesting changes to the new code you wrote!
```

Comment:
For the github owner it can actually be tricky to pull down the forked version and edit it first.  This might be glossy here, but are we gonna cover that?